nsiqcppstyle_exe.py

```diff
@@ -47,123 +46,168 @@
 def ShowMessageAndExit(msg, usageOutput=True):
     console.Err.Error(msg)
     if usageOutput:
-        Usage()
+        get_parser().print_usage()
+        sys.exit(0)
     sys.exit(-1)
 
 
-def Usage():
-    print(
-        """
-======================================================================================
-Usage: nsiqcppstyle [Options]
-           targetdirectory
+def get_parser():
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=title
+        + """Apply custom rules on C/C++ code to report violations of your coding standards
 
 [Example]
    nsiqcppstyle .
    nsiqcppstyle targetdir
    nsiqcppstyle -f filefilterpath targetfilepath
+""",
+        epilog="""
+
+* If options like `--filter-string` and `--var` (which accept one or more arguments) are the last option, remember to separate the arguments and targets by adding a `--` before the targets. Eg:
+  --var key1:value key2:value -- my_target_path
```

Comment:
(1) Would you consider surrounding the key/values with double quotes?  I have always needed it because of the embedded space.

  --var "key1:value key2:value" -- my_target_path

(2) How about inserting a word or two about the embedded space being mandatory?  The command parser will fail if the space is missing.

Reply:
I may be in the past while you're in the present :smiley:

Up until now, a "--var" key/value had a mandatory space between the key and value (e.g., "--var key1:[space]value".  Without the embedded space the parser will not parse the key/value correctly.  I always felt that requiring a space was artificial and clumsy.

I see you new code:

   parser.add_argument(
        "--var",
        action="extend",
        nargs="+",
        help="Provide variables in 'KEY:VALUE' format to customize the rule behavior. Can be used multiple times",
    )

In yout help text above you do not show a space.  Can you tell me if you can pass a "--var" WITHOUT a space and it will parse it correctly?  It would be wonderful if we could get rid of the space.  If you can, then change all "--var" examples and text by eliminating the embedded space.

Reply:
I can confirm you do not need a space after ':' with the new argparse cli. You do still need a space in the config file since I'm not interested in breaking that behavior